modules/neovim/init/spellcheck.nix

```diff
@@ -134,10 +135,17 @@ in {
 
       options = {
         spell = true;
-        spelllang = cfg.languages;
+
+        # Workaround for Neovim's spelllang setup. It can be
+        #  - a string "en"
+        #  - multiple strings, separated with commas
+        # toLuaObject cannot generate the correct type here, unless we take a string here.
+        spelllang = mkLuaInline "${concatStringsSep "," ["en" "de"]}";
```

Comment:
You forgot to use the actual value. Also, mkLuaInline and the string interpolation aren't needed.